Automattic/sensei
# Extend standard Vimeo video embed

### Changes proposed in this Pull Request

* Initialize the Vimeo player in JavaScript and handle events to implement Video-course progression settings.
* I based this solution on [the YouTube implementation](https://github.com/Automattic/sensei/pull/4546), includes all changes from [Video block extension](https://github.com/Automattic/sensei/pull/4562)

### Testing instructions

* Enable the feature flag: `add_filter( 'sensei_feature_flag_video_based_course_progression', '__return_true' );`
  The preferable place is inside `sensei-lms.php`, somewhere in the beginning.
* Run `npm start`
* Enable all video settings for a course.
* Add a Vimeo video to the course lesson.
* Enroll in the course.
* Open the lesson with the Vimeo video.
* Watch the video completely and wait for the auto-submission of the form.
* Disable autocomplete setting and open the lesson again.
* Try to complete the lesson before you watched it completely. Shouldn't submit the form.
* Watch the video completely and wait for the auto-submission of the form. Should work. Also, the complete lesson button should become enabled.
* Test of auto-pause:
  * Start playing video.
  * Try to change tab/window.
  * Video should be paused.
